src/core/task/index.ts

```diff
@@ -1324,20 +1324,27 @@ export class Task {
 			return this.executeCommandInNode(command)
 		}
 
-		// CRITICAL: CLI subagent commands MUST use VSCode terminal mode (not backgroundExec)
-		// Reason: Creates a three-way deadlock when using backgroundExec:
-		//   1. Extension blocks in 'await process' waiting for CLI to exit
-		//   2. CLI blocks waiting for gRPC messages from its child gRPC server
-		//   3. gRPC server (child of CLI) needs extension to process tasks
-		// Solution: Always use VSCode terminal for CLI commands
-		const useVscodeTerminal = isSubagent
+		// Force subagents to use background terminal (hidden execution)
+		const useBackgroundTerminal = isSubagent
```

Comment:
This seems redundant if we are just renaming the a value. We could just use isSubagent instead?

```suggestion
```